src/util/async/context/impl/Execution.hpp

```diff
@@ -42,15 +42,22 @@ struct SpawnDispatchStrategy {
              fn = std::forward<decltype(fn)>(fn)](auto yield) mutable {
                 if constexpr (SomeStoppableOutcome<OutcomeType>) {
                     auto& stopSource = outcome.getStopSource();
-                    fn(outcome, stopSource, stopSource[yield]);
+                    std::invoke(fn, outcome, stopSource, stopSource[yield]);
                 } else {
-                    fn(outcome);
+                    std::invoke(fn, outcome);
                 }
             }
         );
 
         return op;
     }
+
+    template <typename ContextType>
+    static void
+    post(ContextType& ctx, auto&& fn)
```

Comment:
Let's create a typename for `fn` above as well, because now it's not consistent (you have type for `ctx`)
And then use this type in `std::forward` below